modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -177,15 +179,19 @@ def __constructor__(self):
     def __init__(
         self,
         partitions,
-        index,
-        columns,
+        index=None,
+        columns=None,
         row_lengths=None,
         column_widths=None,
         dtypes=None,
     ):
         self._partitions = partitions
-        self._index_cache = ensure_index(index)
-        self._columns_cache = ensure_index(columns)
+        self._index_cache = ensure_index(index) if index is not None else None
+        self._columns_cache = ensure_index(columns) if columns is not None else None
+        self._row_lengths_cache = row_lengths
+        self._column_widths_cache = column_widths
```

Comment:
I assume we should check `self._row_lengths_cache` (`self._column_widths_cache`) instead of `row_lengths` (column_widths) in the conditions below as we do with `self._index_cache` (`self._columns_cache`).

Reply:
they're equal at this point, and current approach is shorter :)

Reply:
I understand they are equal at this point but we should probably unify the use of these things because there might be some postprocessing of these things (either `self._row_lengths_cache` (`self._column_widths_cache`) or `row_lengths` (`column_widths`), either both) and we can lose the right things for the check.

Reply:
ping

Reply:
hmm, postprocessing is a valid point... let me fix that in a sec